hack/check_images.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO="gcr.io/k8s-staging-boskos"
+
+# Check if a commit SHA was provided as a command-line argument
+if [ -z "$1" ]; then
+  echo "Usage: $0 <last_7_chars_commit_sha>"
+  echo "Please provide the last 7 chars of the commit hash to search for in the image tags."
+  exit 1
+fi
+
+COMMIT_SHA="$1"
+
+# Get a list of all images in the specified repository
+images=$(gcloud container images list --repository="$REPO" --format="get(name)")
```

Comment:
Hmm, what if there are random stray images?
I feel like we should identify them by some other means (also this script will only work with GCR/AR and not other repos because of this particular line)